chips/qemu_rv32_virt_chip/src/plic.rs

```diff
@@ -0,0 +1,147 @@
+//! Platform Level Interrupt Control peripheral driver.
+
+use kernel::common::cells::VolatileCell;
+use kernel::common::registers::LocalRegisterCopy;
+use kernel::common::registers::{register_bitfields, ReadWrite};
+use kernel::common::StaticRef;
+
+pub const PLIC_BASE: StaticRef<PlicRegisters> =
+    unsafe { StaticRef::new(0x0c00_0000 as *const PlicRegisters) };
+
+pub static mut PLIC: Plic = Plic::new(PLIC_BASE);
+
+#[repr(C)]
+pub struct PlicRegisters {
+    /// Interrupt Priority Register
+    _reserved0: u32,
+    priority: [ReadWrite<u32, priority::Register>; 51],
+    _reserved1: [u8; 3888],
+    /// Interrupt Pending Register
+    pending: [ReadWrite<u32>; 2],
+    _reserved2: [u8; 4088],
+    /// Interrupt Enable Register
+    enable: [ReadWrite<u32>; 2],
+    _reserved3: [u8; 2088952],
+    /// Priority Threshold Register
+    threshold: ReadWrite<u32, priority::Register>,
+    /// Claim/Complete Register
+    claim: ReadWrite<u32>,
+}
```

Comment:
You can't copy the PLIC from the SiFive chip?

Reply:
To be sure I understand what you're proposing. I should add the sifive chip as a dependency to this and then just use it from there? Sure, I can do that. It's technically slightly weird to have the `qemu_rv32_virt_chip` dependent on this crate but QEMU seems to have a similar-ish architecture (by just reusing the SiFive PLIC implementation).

Reply:
Yep, that's what I'm proposing. I don't think that it's weird. That is why the RISC-V chips are all split up, to allow sharing vendor IP between chips.

QEMU uses the SiFive PLIC internally (which is now almost a RISC-V standard) so it makes sense to reuse it.

Reply:
Okay, I know remember why I've copied it over: this implementation is currently in the `e310x` chip-crate, not in `sifive`, and so I've mirrored that setup. I will move it to the `sifive` crate accordingly and change both chips to use that instead.